core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -264,12 +264,9 @@ private fun getTrackSectionChunkOnWaypoint(
         val endChunk = startChunk + rawInfra.getTrackChunkLength(chunk).distance
         waypointOffset in startChunk..endChunk
     }
-        ?: throw RuntimeException(
-            String.format(
-                "The waypoint is not located on the track section %s",
-                rawInfra.getTrackSectionName(trackSectionId)
-            )
-        )
+        ?: throw OSRDError(ErrorType.InvalidWaypointLocation)
+            .withContext("track", rawInfra.getTrackSectionName(trackSectionId))
```

Comment:
What happens on the other side if you throw this exception without adding the context as you do here? I have a feeling there will be a bug, and I'm assuming that's why we create specific methods such as `newInvalidRollingStockError`. Could you check?